java/base-client/src/main/java/no/bankaxept/epayment/client/base/BaseClient.java

```diff
@@ -51,6 +53,18 @@ public static BaseClient withSuppliedToken(String baseurl, Supplier<String> toke
         return new BaseClient(baseurl, new SuppliedAccessTokenPublisher(tokenSupplier));
     }
 
+    private Map<String, List<String>> filterHeaders(Map<String, List<String>> headers, String correlationId, boolean hasBody) {
+        var filteredHeaders = new LinkedHashMap<>(headers);
+        filteredHeaders.put("X-Correlation-Id", List.of(correlationId));
+        if (!(tokenPublisher instanceof EmptyAccessTokenPublisher))
+            filteredHeaders.put("Authorization", List.of("Bearer " + new AccessTokenSubscriber(tokenPublisher).get(tokenTimeout)));
+        if (hasBody && !headers.containsKey("Content-Type"))
+            filteredHeaders.put("Content-Type", List.of("application/json"));
```

Comment:
Should we reuse the _Content-Type_ from original _headers_? I am thinking this could help with debugging in case it is originally set wrongfully.

Reply:
Nvm. I misread the code!

Reply:
Do you mean they should not be re-used? The clients don't allow the user to set this anyway, this is just in case any of our clients ever need to

Reply:
I mean I misread the code and that this is perfect!